src/Turbo/src/Helper/TurboStream.php

```diff
@@ -86,12 +86,34 @@ public static function refresh(?string $requestId = null): string
         return \sprintf('<turbo-stream action="refresh" request-id="%s"></turbo-stream>', htmlspecialchars($requestId));
     }
 
-    private static function wrap(string $action, string $target, string $html, string $attr = ''): string
+    /**
+     * Custom action and attributes.
+     *
+     * @param array<string, string|int|null> $attr
+     */
+    public static function custom(string $action, string $target, string $html, array $attr = []): string
     {
+        if (\array_key_exists('action', $attr) || \array_key_exists('targets', $attr)) {
+            throw new \InvalidArgumentException('The "action" and "targets" attributes are reserved and cannot be used.');
+        }
+
+        $attr['action'] = $action;
+        $attr['targets'] = $target;
+        $attr = array_merge(['action' => $action, 'targets' => $target], $attr);
+
+        $attrString = '';
+        foreach ($attr as $key => $value) {
+            if (null === $value) {
+                $attrString .= \sprintf(' %s', $key);
```

Comment:
Don't we need to encode keys here?

Reply:
I don't really see a case where the key would have special characters, but it doesn't hurt, so better safe than sorry